app/Integration/Academic/Repository/AcademicQueryBuilderRepository.php

```diff
@@ -36,47 +35,64 @@
     /**
      * @var CrudFacade
      */
-    private $query;
+    private $academicQuery;
     /**
      * @var DatabaseManager
      */
     private $databaseManager;
+    /**
+     * @var FromAcademicMapper
+     */
+    private $fromAcademicMapper;
+    /**
+     * @var FromArrayMapper
+     */
+    private $fromArrayMapper;
+    /**
+     * @var CrudFacade
+     */
+    private $articleQuery;
 
     public function __construct(
         DatabaseManager $databaseManager,
-        CrudFacade $query,
-        AcademicMapper $academicMapper,
+        CrudFacade $academicQuery,
```

Comment:
Too many dependencies, that's a bad smell. We have to think a way to simplify this class

Reply:
Imho the crud interface can be dropped.
It's useless in this context.

Reply:
As I said, I think we have to rethink a little repositories. For now we will keep them